lib/location/scan_cellular.c

```diff
@@ -24,11 +24,28 @@ static struct lte_lc_cells_info scan_cellular_info = {
 };
 
 static bool running;
+static bool timeout_occurred;
 /* Indicates when individual ncellmeas operation is completed. This is internal to this file. */
 static struct k_sem scan_cellular_sem_ncellmeas_evt;
 /* Requested number of cells to be searched. */
 static int8_t scan_cellular_cell_count;
 
+/** Handler for timeout. */
+static void scan_cellular_timeout_work_fn(struct k_work *work);
+
+/** Work item for timeout handler. */
+K_WORK_DELAYABLE_DEFINE(scan_cellular_timeout_work, scan_cellular_timeout_work_fn);
+
+/**
+ * Handler for backup timeout, which ensures we won't be waiting for LTE_LC_EVT_NEIGHBOR_CELL_MEAS
+ * event forever after lte_lc_neighbor_cell_measurement_cancel() in case it is never sent by
+ * lower layers due to a bug.
```

Comment:
I would rephrase the ending to something like "in case it would never be sent". The current comment could be understood so, that there is a bug, because of which the event is sometimes not sent. Which I don't think is the case?